packages/core-components/src/lib/unsorted/viz/BarChart.svelte

```diff
@@ -31,7 +31,7 @@
 	export let yMin = undefined;
 	export let yMax = undefined;
 	export let swapXY = false;
-	export let showAllXAxisLabels = xType === 'category';
+	export let showAllXAxisLabels = xType === 'category' || typeof data?.[0]?.[x] === "string";
```

Comment:
are we OK to rely on typeof here, instead of the evidence type shipped with the data object?